cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -147,10 +176,12 @@ parse_numeric(const char* data, long start, long end, ParseOptions const& opts)
   bool all_digits_valid = true;
 
   // Handle negative values if necessary
-  int32_t sign = 1;
-  if (data[start] == '-') {
-    sign = -1;
-    start++;
+  int32_t sign = 1 - (2 * (data[start] == '-'));
```

Comment:
The old code was more readable. Perhaps a ternary would allow you to do the same thing.

Reply:
Ternary might introduce a branch if it doesn't get compiled out. I have not verified this statement.

Reply:
That was me trying to avoid branching, gave up readability. Used ternary operator now.